common/persistence/execution_manager.go

```diff
@@ -307,13 +322,22 @@ func (m *executionManagerImpl) CreateWorkflowExecution(
 	if _, err := m.persistence.CreateWorkflowExecution(newRequest); err != nil {
 		return nil, err
 	}
-	return &CreateWorkflowExecutionResponse{HistorySize: newHistoryDiff}, nil
+	return &CreateWorkflowExecutionResponse{
+		NewMutableStateStats: *statusOfInternalWorkflowSnapshot(
+			serializedNewWorkflowSnapshot,
+			newHistoryDiff,
+		),
+	}, nil
 }
 
 func (m *executionManagerImpl) serializeWorkflowEventBatches(
 	shardID int32,
 	eventBatches []*WorkflowEvents,
 ) ([]*InternalAppendHistoryNodesRequest, int64, error) {
+	if len(eventBatches) == 0 {
+		return nil, 0, nil
+	}
+
```

Comment:
no need this check, for loop handles it already.

Reply:
additional mem allocation, e.g. the `make(...)`

Reply:
it is just empty slice in exception case, shouldn't be a concern.